common/common/config/config.py

```diff
@@ -64,6 +67,7 @@ def __init__(self, config_overrides: dict | None = None, **kwargs):
             kwargs = merge_flattened_into_nested(flattened_overrides, kwargs)
 
         super().__init__(**kwargs)
+        self.custom_validate()
```

Comment:
Remove this and use pydantic validation - https://docs.pydantic.dev/latest/concepts/validators/#model-validators